overlays/eval-on-current.nix

```diff
@@ -2,7 +2,15 @@ final: prev: {
   # This overlay makes `evalPackages` is like `buildPackages`, but on
   # `builtins.currentSystem`.
   # See `eval-on-build.nix` for alternative that just uses `buildPackages`.
-  evalPackages = (import final.path (import ../. {
-    inherit (final.haskell-nix) defaultCompilerNixName;
-  }).nixpkgsArgs).buildPackages;
+  evalPackages = (import final.path (import ../. ({
+      # If check materialization was turned on also check in eval packages
+      inherit (final.haskell-nix) checkMaterialization;
+      # Forward any sourcesOverrides by simply passing all the sources as sourcesOverrides
+      sourcesOverride = final.haskell-nix.sources;
+    } //
+      # Forward user specified defaultCompilerNixName
```

Comment:
:see_no_evil:

Reply:
I think we might be able to make this nicer if we save the overlays used in say `haskell-nix.overlays` and then just reapply them here.

Reply:
Or be much more aggressive about dropping `defaultCompilerNixName` :angel:

Reply:
I've updated this so overlays are all applied without needing recreate them this way.  Also `defaultCompilerNixName` should always result in a warning when it is used (so we can remove it once we are happy it will not cause disruption).